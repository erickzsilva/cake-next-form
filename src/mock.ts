import { CakeResponse } from './sharedTypes/types'

export const mock: CakeResponse = {
  title: 'Cake Order Form',
  description: `Order your freshly baked cakes using only the finest quality natural ingredientes.`,
  textChooseYourCake:
    'Please choose your favorite cake from among the following',
  subTitle: 'Order Information',
  cakes: [
    {
      value: 'Red Fruit Cake',
      name: 'redFruitCake',
      alt: 'Red Fruit Cake',
      srcImage: 'img/redFruitCake.jpg'
    },
    {
      value: 'Strawberry Sponge Cake',
      name: 'strawberrySpongeCake',
      alt: 'Strawberry Sponge Cake',
      srcImage: 'img/strawberrySpongeCake.jpg'
    },
    {
      value: 'Hazelnut Pie',
      name: 'hazelnutPie',
      alt: 'Hazelnut Pie',
      srcImage: 'img/hazelnutPie.jpg'
    },
    {
      value: 'MM Cake',
      name: 'mmCake',
      alt: 'MM Cake',
      srcImage: 'img/mmCake.jpg'
    }
  ],
  country: ['Argentina', 'Brasil', 'Fran√ßa', 'Mexico', 'Paraguai', 'Portugal']
}

export const mockHead = {
  title: 'Cake Order Form',
  description: `Order your freshly baked cakes using only the finest quality natural ingredientes.`
}
