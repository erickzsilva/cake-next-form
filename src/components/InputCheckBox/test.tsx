import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import InputChebox from '.'
import { mock } from '../../mock'

describe('<InputChebox />', () => {
  it('should render the checkbox and on click it should be selected', () => {
    const { container } = render(
      <InputChebox
        alt={mock.cakes[0].alt}
        srcImage={mock.cakes[0].srcImage}
        value={mock.cakes[0].value}
        defaultChecked={false}
      />
    )

    expect(screen.getByTestId('checkbox')).toBeInTheDocument()

    expect(userEvent.click(screen.getByTestId('checkbox'))).toBeChecked

    expect(container.firstChild).toMatchSnapshot()
  })
})
