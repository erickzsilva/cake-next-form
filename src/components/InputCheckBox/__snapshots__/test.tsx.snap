// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputChebox /> should render the checkbox and on click it should be selected 1`] = `
.c0 {
  align-items: center;
  display: flex;
  justify-content: center;
  max-width: 180px;
}

.c1 {
  margin-right: 8px;
  position: relative;
}

.c2 {
  width: 15px;
  height: 15px;
  border-radius: 50%;
  vertical-align: middle;
  border: 1px solid #eba70e;
  appearance: none;
  -webkit-appearance: none;
  outline: none;
  cursor: pointer;
}

.c2:checked {
  background: url('img/check.png') no-repeat center center;
  background-size: 17px;
  width: 17px;
  height: 17px;
}

.c2:invalid {
  border-color: red;
}

.c3 {
  height: 100px;
  max-width: 150px;
  width: 100%;
}

@media screen and (max-width: 768px) {
  .c0 {
    max-width: 300px;
  }
}

@media screen and (max-width: 768px) {
  .c2 {
    background-size: 17px;
    width: 17px;
    height: 17px;
  }
}

@media screen and (max-width: 768px) {
  .c3 {
    height: unset;
    max-width: unset;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <input
      class="c2"
      data-testid="checkbox"
      type="checkbox"
      value="Red Fruit Cake"
    />
  </div>
  <img
    alt="Red Fruit Cake"
    class="c3"
    src="img/redFruitCake.jpg"
  />
</div>
`;
